### ISSUE TITLE
Ruby 2.7 Warning: Using the last argument as keyword parameters is deprecated

### ISSUE THREAD

**Issue by AntonSynytsia** [SketchUp/api-issue-tracker#578]
I am using `view.draw` and this function causes the following warning message to display in console:
> warning: Using the last argument as keyword parameters is deprecated

I believe this is a Ruby 2.7-related deprecation message.

This is the arguments I pass to the function:
```rb
view.draw(GL_TRIANGLES, triplets, draw_opts)
```

**Comment by thomthom** [comment]
Yea, Ruby 2.7 had some changes in that respect, preparing people for Ruby 3.

I think an alternative is: `view.draw(GL_TRIANGLES, triplets, **draw_opts)`

**Comment by DanRathbun** [comment]
Ie: The "headlights" or double-splat operator will expand your hash into a parameter list.

@thomthom Did you prepare a Ruby 2.7 breaking changes primer this year ? I though I saw it somewhere. ;)

**Comment by MSP-Greg** [comment]
I apologize for not remembering it.  Ruby 2.7.2 reverted the warning.  But, 3.0 is currently behaving differently.

Below is code and output from Ruby trunk, 2.7.2, & 2.7.1.

```
puts RUBY_DESCRIPTION

def draw(openglenum, *points, **options)
  puts points.inspect, options.inspect
end

draw 1.0, 2.0, 3.0, {help: 'help'}

————————————————————————————————————————————————————————————————————————————————

ruby 3.0.0dev (2020-11-30T10:06:25Z master 89774a938a) [x64-mingw32]
[2.0, 3.0, {:help=>"help"}]
{}


ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x64-mingw32]
[2.0, 3.0]
{:help=>"help"}


ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x64-mingw32]
keywords.rb:7: warning: Using the last argument as keyword parameters is deprecated;
  maybe ** should be added to the call
keywords.rb:3: warning: The called method `draw' is defined here
[2.0, 3.0]
{:help=>"help"}
```

**Comment by AntonSynytsia** [comment]
Thank you for your insights everyone!

Wondering why Sketchup didn't choose the latest 2.7.x when releasing

Also, great job about 3.0. Although I feel a little nervous about having so many Rubies. To support SU2016 there will need to be at 5 binaries per platform: 2.0 2.2. 2.5. 2.7 3.0. It would look a little overwhelming. Not a problem though.

**Comment by MSP-Greg** [comment]
@AntonSynytsia

Apologies, I wasn't clear.  Ruby 3.0 isn't released yet, and I suspect that Trimble/SketchUp won't be releasing it in SketchUp until 2023 or later (just a guess).  It does have some new features.

I was trying to point out that the handling of keyword arguments is changing, largely due to the mess of how hash and keyword arguments have been kind of mixed together in 2.x versions...

**Comment by thomthom** [comment]
> Wondering why Sketchup didn't choose the latest 2.7.x when releasing

Ruby 2.7.1 was the latest when we were doing the work. 2.7.2 was released October 2nd. There was no capacity to update this late as it would require a lot of re-testing.

> Also, great job about 3.0. Although I feel a little nervous about having so many Rubies. To support SU2016 there will need to be at 5 binaries per platform: 2.0 2.2. 2.5. 2.7 3.0. It would look a little overwhelming. Not a problem though.

I know the feeling. But at the same time, for extensions being maintained I'm starting to easily drop support for versions older than SU2017. Significant changes since then and declining amount of users. Old versions compatible with older versions are left available though.

**Comment by DanRathbun** [comment]
> I was trying to point out that the handling of keyword arguments is changing, largely due to the mess of how hash and keyword arguments have been kind of mixed together in 2.x versions...

I liked how they were "bundled" together with the hash args myself. I never had any issues with this.

Bundling them up with the positional arguments has a better chance of breaking changes. 
What if the API functions are not written to support the additional hash argument as part of the positional args ?

**Comment by DanRathbun** [comment]
**FYI**

> Ruby 2.7.1 was the latest when we were doing the work.

... and so SU2021.0 released with Ruby **2.7.1**

> Ruby 2.7.2 reverted the warning.

SketchUp 2021.1 was upgraded to Ruby **2.7.2**
